core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -112,19 +113,43 @@ class InvokerReactive(
   private val ack = (tid: TransactionId,
                      activationResult: WhiskActivation,
                      blockingInvoke: Boolean,
-                     controllerInstance: InstanceId) => {
+                     controllerInstance: InstanceId,
+                     userId: UUID) => {
     implicit val transid: TransactionId = tid
 
     def send(res: Either[ActivationId, WhiskActivation], recovery: Boolean = false) = {
       val msg = CompletionMessage(transid, res, instance)
-
       producer.send(s"completed${controllerInstance.toInt}", msg).andThen {
         case Success(_) =>
           logging.info(
             this,
             s"posted ${if (recovery) "recovery" else "completion"} of activation ${activationResult.activationId}")
       }
     }
+    // Potentially sends activation metadata to kafka if user events are enabled
+    UserEvents.send(
+      producer, {
+        val activation = Activation(
```

Comment:
this `Activation` type is better served as a method in `WhiskActivation` instead.